hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanFlowStatusTask.groovy

```diff
@@ -0,0 +1,248 @@
+
+      if (fetchFlowsResponse.toLowerCase().contains("error")) {
+        // Check if session has expired. If so, re-login.
+        if (fetchFlowsResponse.toLowerCase().contains("session")) {
+          logger.lifecycle("\nPrevious Azkaban session expired. Please re-login.");
+          getAzkabanFlowStatus(null);
+        } else {
+          // If response contains other than session error
+          logger.error("Fetching flows from " + azkProject.azkabanUrl + " failed. Reason: " + new JSONObject(fetchFlowsResponse).get("error"));
+        }
+        return;
+      }
+
+      List<String> flows = AzkabanHelper.fetchSortedFlows(new JSONObject(fetchFlowsResponse));
+      if (flows.isEmpty()) {
+        logger.lifecycle("No flows defined in current project");
+        return;
+      }
+
+      //Pool HTTP Get requests for getting most recent ExecID for each flow
+      List<String> responseList = AzkabanClient.batchfetchLatestExecution(azkProject.azkabanUrl, azkProject.azkabanProjName, flows, sessionId);
+
+      List<String> execIds = new ArrayList<String>();
+      for (String execResponse : responseList) {
+        JSONObject execIDResponse = new JSONObject(execResponse);
+        if (execIDResponse.has("error")) {
+          logger.error("Could not get flow due to " + execIDResponse.get("error").toString());
+          return;
+        } else if (execIDResponse.has("executions")) {
+          JSONArray lastExecArray = execIDResponse.getJSONArray("executions");
+          if (lastExecArray != null && !lastExecArray.isNull(0)) {
+            JSONObject lastExecObj = new JSONObject(lastExecArray.get(0).toString());
+            execIds.add(lastExecObj.get("execId").toString());
+          }
+        }
+      }
+
+      if (!execIds.isEmpty()) {
+        //Pool HTTP Get requests for getting exec Job statuses
+        responseList = AzkabanClient.batchfetchFlowExecution(azkProject.azkabanUrl, execIds, sessionId);
+
+        if (project.hasProperty("flow")) {
+          if (project.getProperties().get("flow").toString().isEmpty()) {
+            jobStatus(flows, responseList);
+          } else {
+            List<String> sortedFlowSet = new ArrayList(new HashSet<String>(Arrays.asList(project.getProperties().get("flow").toString().split(",+"))));
+            Collections.sort(sortedFlowSet);
+            jobStatus(sortedFlowSet, responseList);
+          }
+        } else {
+          flowStatus(flows, responseList);
+        }
+
+      } else {
+        logger.lifecycle("Project ${azkProject.azkabanProjName} has no flow previously executed.");
+      }
+
+    } catch (Exception ex) {
+      logger.info("Could not fetch flow status. ${ex.getMessage()}");
+    }
+  }
+
+  /**
+   * Prints the Job Status of the latest flow executions.
+   *
+   * @param flowSet
+   * @param responseList
+   */
+  void jobStatus(List<String> flowSet, List<String> responseList) {
```

Comment:
It looks like we can move the jobStatus and flowStatus methods to a helper class in the azkaban-cli project